src/vcpkg/binarycaching.cpp

```diff
@@ -1160,6 +1306,36 @@ namespace
                 handle_readwrite(
                     state->url_templates_to_get, state->azblob_templates_to_put, std::move(p), segments, 3);
             }
+            else if (segments[0].second == "x-gcs")
+            {
+                // Scheme: x-gsutil,<prefix>[,<readwrite>]
+                if (segments.size() < 2)
+                {
+                    return add_error("expected arguments: binary config 'gcs' requires at least a prefix",
+                                     segments[0].first);
+                }
+
+                if (!Strings::starts_with(segments[1].second, "gs://"))
+                {
+                    return add_error(
+                        "invalid argument: binary config 'gcs' requires an gs:// base url as the first argument",
+                        segments[1].first);
+                }
+
+                if (segments.size() > 3)
+                {
+                    return add_error("unexpected arguments: binary config 'gcs' requires 1 or 2 arguments",
+                                     segments[4].first);
```

Comment:
maybe do something similar to `handle_segments()` for more helpful error messages

Reply:
This is consistent with all the other binary configs.  Happy to change it, but do we want this one to be very different?